### PR TITLE
Add promoted snaps endpoint

### PR DESCRIPTION
**Pull request by tbille**
# Summary

- Add promoted snaps endpoint

# QA 

- Checkout to this branch
- `./run`
- Go to http://0.0.0.0:8004/snaps/
- See promoted snaps ;)

# Problem...
I don't know how to center a title with vanilla...

### CODE REVIEW

File: templates/promoted.html [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{% extends "_layout.html" %}
+
+{% block title %}
+  Promoted Snaps — Linux software in the Snap Store
+{% endblock %}
+
+{% block content %}
+  <section class="p-strip is-shallow">
+      <div class="col-12">
```
Comment (bartaz) [inline templates/promoted.html]: There should be a `<div class="row">` around that col.

File: templates/promoted.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          <h3>Featured snaps</h3>
```
Comment (bartaz) [inline templates/promoted.html]: These are not featured snaps. Do we need a title here?

File: app.py [hunk 3]
```diff
@@ -204,6 +222,14 @@ def discover():
     )
 
 
+@app.route('/promoted/')
```
Comment (bartaz) [inline app.py]: Is there meant to be this separate page for those, or should 'promoted' snaps be displayed on some other existing page (search, discover?)? @Greggless?

File: templates/promoted.html (hunk 2 above)
Comment (Greggless) [reply templates/promoted.html]: We won't need a title

File: app.py (hunk 3 above)
Comment (Greggless) [reply app.py]: On phase 2 of /discover (https://github.com/CanonicalLtd/snapcraft-design/issues/197) we link to a page using `see more snaps`. This is the core of that page, a listing of all 'promoted snaps. 

It needs some design love, mainly including a search in the hero. @tbille put this together quickly last night (thanks!)

File: templates/promoted.html [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+{% block title %}
+  Promoted Snaps — Linux software in the Snap Store
+{% endblock %}
+
+{% block content %}
+  <section class="p-strip is-shallow">
+      <div class="row">
+          <div class="col-12">
+              <h1>Promoted snaps</h1>
```
Comment (bartaz) [inline templates/promoted.html]: I think @Greggless mentioned we don't need title at all on this page.

File: templates/promoted.html [hunk 5]
```diff
@@ -0,0 +1,40 @@
+{% extends "_layout.html" %}
+
+{% block title %}
+  Promoted Snaps — Linux software in the Snap Store
```
Comment (bartaz) [inline templates/promoted.html]: I'm not sure - is 'promoted snaps' official way how they are called? Is that the title we want for that page?

**Comment by nottrobin** [comment]
@tbille is this linked to an issue? I'm trying to track down where the decision to call this "promoted" came from, to work out who I should be bring up this point to:

I raised an issues with that wording, that "promoted" is semantically very similar to "featured" and will almost certainly confuse people both internally and externally. I'd prefer an adjective like "approved" or something, but maybe for the URL endpoint we could use something entirely neutral like simply "/snaps/".

**Comment by Greggless** [comment]
I believe it had been specified in the [sections and promoted](https://docs.google.com/document/d/1J5a0W92RbuKeGNrYmaTbzXqbE0NpQ8e0xnJG6D4MLVA/edit) documentation but I don't know its true origin.

I agree with @nottrobin  that these shouldn't be called 'promoted' snaps, and have no objection to using 'approved'.

Having the URL as `https://snapcraft.io/snaps` makes sense to me. It'd look something like [this](https://docs.google.com/document/d/138XzmkXhjuS7pVSR0geEpgQbUWOrsHLFsh9hrr6zm_E/edit) until we decide to implement ~~categories~~ sections, filters and sorting as specified in **Phase 3** of the [phased rollout of /discover](https://docs.google.com/a/canonical.com/spreadsheets/d/1Mn9WfpsHA5KfrM_c4VuBpKD44khIph0Q_FPS70cY45M/edit?usp=sharing)

File: app.py (hunk 3 above)
Comment (nottrobin) [reply app.py]: This should be changed to `/snaps/`.

**Comment by Greggless** [comment]
# MERGEDAT  🚀 :shipit:

We will use this endpoint when we come to complete phase 2 of /discover plan

**Pull request merged**